pkgs/desktops/cinnamon/cinnamon-common/default.nix

```diff
@@ -52,6 +53,23 @@
 , perl
 }:
 
+let
+  pythonEnv = python3.withPackages (pp: with pp; [
+    dbus-python
+    setproctitle
+    pygobject3
+    pycairo
+    python3.pkgs.xapp # The scope prefix is required
```

Comment:
You are really sure on this? Not just xapp?

Reply:
Yes, I tested this on [one of the PR](https://github.com/NixOS/nixpkgs/pull/189824#discussion_r962701929) before, we have both `cinnamon.xapp` and `python3.pkgs.xapp`, they are different package but are both needed here and we have `xapp` as one of the args (for `cinnamon.xapp`) and looks like the arg will have higher priority if we just use `xapp` here. I am unfortunately not able to explain the reason :-(

I wonder if we should just rename `python3.pkgs.xapp` to `python3.pkgs.python-xapp` according to upstream's [meson project name](https://github.com/linuxmint/python3-xapp/blob/master/meson.build#L1).

Reply:
We can also use `pp.xapp` here.